RFCs/RFC1/RFC.txt

```diff
@@ -0,0 +1,121 @@
+   encoded data into text files. The current object storage service[CIAOS]
+   performs upload, get, update and delete functionalities. The upload 
+   operation excepts a key and a base64 encoded file from the client to be
+   uploaded and stores the base64 encoded content inside a text file in 
+   directory with key name as folder name so that when get operation is 
+   performed the retrieval of encoded content is from the directory with 
+   the key name.
+
+   The haystack implementation does not use any encoding of files, the files
+   are directly uploaded and the haystack implementation writes each file 
+   into a bin file and stores the offset[the starting address to the file]
+   and the size of the file. So that while retrieving the storage service 
+   uses the offset and size value to map the file, read and provide download
+   option to client 
+
+ 
+
+Testing: 
+
+   The test was run on Lenovo Legion Y740-I5RH  
+   System Specifications: i7-9750H [cores:6, hreads12, max: 4.50 GHz, 
+   base: 2.60 GHz], 16gb ram, GTX 1660 ti, 512gb SSD,1TB HHD. 
+
+   The dummy file used for testing purpose is “x64n.rpf” which is 1.6gb 
+   in size, the image testing.jpg in the repository shows the testing 
+   time taken to upload as well as download the given file on a haystack
+   storage model, which was approximately 19 seconds .The same test 
+   couldn’t be performed on Current Storage service as it crashed the 
+   system due to high CPU usage while base64 encoding of dummy file.
+
+   Another test was performed to measure the same time taken but over a 
+   30mb dummy file. The images CIAOS_testing.jpg and haystack_testing.jpg 
+   show the time taken to download and upload the dummy file over current
+   storage service implementation and Haystack implementation respectfully.
+   Here we can clearly see that the haystack takes at most 0.35 seconds 
+   while the CIAOS implementation takes atleast 5 seconds to perform the 
+   same operation.
+   
+   The key differences between Haystack implementation and CIAOS is that 
+   the haystack clearly outperforms CIAOS in terms of speed, system resource
+   usage as well as the metadata. The above testing shows that haystack 
+   clearly outperforms CIAOS in terms of speed, the CIAOS implementation used
+   high system resources as it had to store really long base64 encoded data 
+   in a number of text files and also open each text file and read the base64
+   content to write it to a list and than provide the user with encoded data.
+   This clearly required more system resources to perform each operation as 
+   well as created unecessarily more metadata for the system to store.
+   Therefore wastage of disk space which can intern be carefully utilised by 
+   haystack.
+
+
+ 
+
+Alternative to Current implementation of Storing Metadata in haystack:
+
+   To store keys,offset and size of upload it would be more efficient to 
```

Comment:
Don't mention JSON. JSON is something I used for my easyness. Haystack paper never mentions JSON. As I said we should always compare what's the paper not what I implemented. So you can remove this section not required